sc/src/test/java/uk/ac/ebi/atlas/tsne/TSnePlotServiceDaoIT.java

```diff
@@ -103,6 +111,24 @@ void testPerplexities(String experimentAccession) {
                 .doesNotHaveDuplicates();
     }
 
+    @ParameterizedTest
+    @MethodSource("randomExperimentAccessionProvider")
+    void testNumberOfCellsByExperimentAccession(String experimentAccession) {
+        cleanDatabaseTables();
```

Comment:
You don't need to call `cleanDatabaseTables` and `populateDatabaseTables` yourself. That what the `@AfterAll` and `@BeforeAll` annotations on those methods are for. (see [here](https://junit.org/junit5/docs/current/user-guide/#writing-tests-annotations) for a description of all the possible JUnit 5 annotations)

Reply:
We need to call `cleanDatabaseTables` and `populateDatabaseTables` in this case, since we are using different fixure.sql dataset. We need to mount a new dataset and clear it before running the following tests. `@AfterAll` and `@BeforeAll` annotations will only run once before and after all the tests, which may not help in this case.

Reply:
Is there any reason you can't use the same fixtures for all tests? Or you could change `@AfterAll` and `@BeforeAll` to `@AfterEach` and `@BeforeEach`.

Reply:
In this test, we test the count of cells. To make a comprehensive test we count the lines of the local file to match the return result by querying in the fixture. If the fixture is a partition of the full dataset, then it will fail.
Instead of changing to @AfterEach and @AfterEach, we could keep the testing speed as much as we can, since we only need to reload the dataset once.

Reply:
@lingyun1010 Could you add a short comment at the top of this method explaining why you need a particular fixture for this test?